doc/book/usage.md

```diff
@@ -114,32 +115,37 @@ use Zend\Db\Sql\Select;
 use Zend\Paginator\Adapter\DbSelect;
 use Zend\Paginator\Paginator;
 
-class MyDbSelect extends DbSelect
-{
-    public function count()
-    {
-        if ($this->rowCount) {
-            return $this->rowCount;
-        }
-
-        $select = new Select();
-        $select
-          ->from('item_counts')
-          ->columns(['c'=>'post_count']);
-
-        $statement = $this->sql->prepareStatementForSqlObject($select);
-        $result    = $statement->execute();
-        $row       = $result->current();
-        $this->rowCount = $row['c'];
-
-        return $this->rowCount;
-    }
-}
-
-$adapter = new MyDbSelect($query, $adapter);
```

Comment:
I'm going to keep this section above, as it's still an option, and because, for users prior to 2.9.0 (where this feature will drop), it's the _only_ option.

I can make that change on merge, however.